lib/credentials/remote_credentials.js

```diff
@@ -0,0 +1,248 @@
+      secretAccessKey: credData.secretAccessKey || credData.SecretAccessKey,
+      sessionToken: credData.sessionToken || credData.Token,
+      expireTime: new Date(credData.expiration || credData.Expiration)
+    };
+  },
+
+  /**
+   * @api private
+   */
+  request: function request(url, callback) {
+    var httpRequest = new AWS.HttpRequest(url);
+    httpRequest.method = 'GET';
+    httpRequest.headers.Accept = 'application/json';
+    var token = this.getRemoteAuthToken();
+    if (token) {
+      httpRequest.headers.Authorization = token;
+    }
+    AWS.util.handleRequestWithRetries(httpRequest, this, callback);
+  },
+
+  /**
+   * @api private
+   */
+  refreshQueue: [],
+
+  /**
+   * Loads the credentials from the relative URI specified by container
+   *
+   * @callback callback function(err)
+   *   Called when the request to the relative URI responds (or fails). When
+   *   this callback is called with no error, it means that the credentials
+   *   information has been loaded into the object (as the `accessKeyId`,
+   *   `secretAccessKey`, `sessionToken`, and `expireTime` properties).
+   *   @param err [Error] if an error occurred, this value will be filled
+   * @see get
+   */
+  refresh: function refresh(callback) {
+    var self = this;
+    var refreshQueue = self.refreshQueue;
+    if (!callback) callback = function(err) { if (err) throw err; };
+    refreshQueue.push({
+      provider: self,
+      errCallback: callback
+    });
+    if (refreshQueue.length > 1) { return; }
+
+    function callbacks(err, creds) {
+      var call, cb;
+      while ((call = refreshQueue.shift()) !== undefined) {
+        cb = call.errCallback;
+        if (!err) AWS.util.update(call.provider, creds);
+        cb(err);
+      }
+    }
+
+    if (process === undefined) {
+      callbacks(AWS.util.error(
+        new Error('No process info available'),
+        { code: this.providerName + 'ProviderFailure' }
+      ));
+      return;
+    }
+    var fullUri = this.getRemoteFullUri();
+    if (fullUri === undefined) {
+      callbacks(AWS.util.error(
+        new Error('Variable ' + this.envRelativeUri + ' or ' + this.envFullUri +
+          ' must be set to use AWS.' + this.providerName + '.'),
+        { code: this.providerName + 'ProviderFailure' }
+      ));
+      return;
+    }
+
+    this.request(fullUri, function(err, data) {
+      if (!err) {
+        try {
+          data = JSON.parse(data);
+          var creds = self.formatCreds(data);
+          if (!self.credsFormatIsValid(creds)) {
+            throw AWS.util.error(
+              new Error('Response data is not in valid format'),
+              { code: this.providerName + 'ProviderFailure' }
+            );
+          }
+        } catch (dataError) {
+          err = dataError;
+        }
+      }
+      callbacks(err, creds);
+    });
+  }
+});
+
+AWS.ECSCredentials = AWS.util.inherit(AWS.RemoteCredentials, {
```

Comment:
Isn't this a duplicate of what's in `lib/credentials/ecs_credentials.js`?

Reply:
Thanks for the catch. I should have removed that when I put it in ecs_credentials.js. Added fix to PR.